ts/src/header-validator/validate-json.ts

```diff
@@ -725,6 +725,102 @@ const priorityField: StructFields<Priority> = {
   priority: field('priority', int64, 0n),
 }
 
+function isDebugTypeSupported(ctx: RegistrationContext, s: string) {
+  if (ctx instanceof SourceContext) {
+    return constants.sourceAggregateDebugTypes.includes(s)
+  } else {
+    return constants.triggerAggregateDebugTypes.includes(s)
+  }
+}
+
+function aggregateDebugType(ctx: RegistrationContext, j: Json): Maybe<string> {
+  return string(ctx, j).peek((s) => {
+    if (!isDebugTypeSupported(ctx, s)) {
+      ctx.warning('unknown type')
+    }
+  })
+}
+
+export type KeyPiece = {
+  keyPiece: bigint
+}
+
+const keyPieceField: StructFields<KeyPiece> = {
+  keyPiece: field('key_piece', hex128),
+}
+
+export type AggregateDebugReportingData = KeyPiece & {
+  types: string[]
+  value: number
+}
+
+function aggregateDebugReportingData(
+  ctx: RegistrationContext,
+  j: Json
+): Maybe<AggregateDebugReportingData> {
+  return struct(ctx, j, {
+    types: field('types', (ctx, j) =>
+      array(ctx, j, aggregateDebugType, { minLength: 1 })
+    ),
+    value: field('value', aggregatableValue),
+
+    ...keyPieceField,
+  })
+}
+
+function aggregateDebugReportingDataList(
+  ctx: RegistrationContext,
+  j: Json
+): Maybe<AggregateDebugReportingData[]> {
+  return array(ctx, j, aggregateDebugReportingData).filter((data) => {
+    const types = new Set<string>()
+    const dups = new Set<string>()
+    for (const d of data) {
+      for (const t of d.types) {
+        if (!isDebugTypeSupported(ctx, t)) {
+          continue
+        }
+        if (types.has(t)) {
+          dups.add(t)
+        } else {
+          types.add(t)
+        }
+      }
+    }
+    if (dups.size > 0) {
+      ctx.error(`duplicate type: ${Array.from(dups).join(', ')}`)
+      return false
+    }
+    return true
+  })
+}
+
+function aggregationCoordinatorOriginField(
+  ctx: RegistrationContext,
+  j: JsonDict
+): Maybe<string> {
+  return field(
+    'aggregation_coordinator_origin',
+    aggregationCoordinatorOrigin,
+    ctx.vsv.aggregationCoordinatorOrigins[0]
+  )(ctx, j)
+}
+
+export type AggregateDebugReportingConfig = KeyPiece & {
+  aggregationCoordinatorOrigin: string
+  data: AggregateDebugReportingData[]
+}
+
+const aggregateDebugReportingConfig: StructFields<
+  AggregateDebugReportingConfig,
+  RegistrationContext
+> = {
+  aggregationCoordinatorOrigin: aggregationCoordinatorOriginField,
+  data: field('data', aggregateDebugReportingDataList, []),
```

Comment:
Isn't this field called `debug_data` in the spec?

Reply:
Yes, also realized this and updated. Thanks.